File: src/main/webapp/open-pricing-tracker/src/app/shared/models/jenkins-job.model.ts
import {autoserialize, autoserializeAs} from "cerialize";
/**
 * Provides the model that described jenkins job
 */
export class JenkinsJobModel {

  public dateNow: number = Date.now();

  public dateStartedAt: Date;
  public _startedAt: number;

  @autoserialize
  public name: string;

  @autoserialize
  public displayName: string;

  @autoserialize
  public buildNumber: number;

  @autoserialize
  public duration: number;

  @autoserialize
  public estimateDuration: number;

  @autoserialize
  public result: string;

  @autoserialize
  public set startedAt(value: number) {
    this._startedAt = value;
    this.dateStartedAt = new Date(value);
  }

  public get startedAt(): number {
    return this._startedAt;
  }

  @autoserialize
  public url: string;

  @autoserialize
  public folder: string;

}

File: src/main/java/com/tracker/domain/jenkins/JenkinsJobModel.java
package com.tracker.domain.jenkins;

import lombok.Getter;
import lombok.Setter;

/**
 * Provides the model of jenkins job
 */
@Getter
@Setter
public class JenkinsJobModel {

    private String name;
    private String displayName;
    private int buildNumber;
    private long duration;
    private long estimateDuration;
    private String result;
    private long startedAt;
    private String url;
    private JenkinsFolder folder;

    public long getFinishedAt() {
        return this.startedAt + this.duration;
    }
}
